app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/state/MessageComposerState.kt

```diff
@@ -166,12 +169,22 @@ data class MessageComposerState(
     fun hideAttachmentOptions() = changeAttachmentOptionsVisibility(false)
     private fun changeAttachmentOptionsVisibility(newValue: Boolean) {
         (messageComposeInputState as? MessageComposeInputState.Active)?.let { activeState ->
-            (activeState.type as? MessageComposeInputType.NewMessage)?.let { newMessageType ->
-                messageComposeInputState = activeState.copy(
-                    type = newMessageType.copy(
-                        attachmentOptionsDisplayed = newValue
+            when (val currentType = activeState.type) {
+                is MessageComposeInputType.NewMessage -> {
+                    messageComposeInputState = activeState.copy(
+                        type = currentType.copy(
+                            attachmentOptionsDisplayed = newValue
+                        )
+                    )
+                }
+                is MessageComposeInputType.SelfDeletingMessage -> {
+                    messageComposeInputState = activeState.copy(
+                        type = currentType.copy(
+                            attachmentOptionsDisplayed = newValue
+                        )
```

Comment:
```suggestion
                is MessageComposeInputType.NewMessage,
                is MessageComposeInputType.SelfDeletingMessage -> {
                    messageComposeInputState = activeState.copy(
                        type = currentType.copy(
                            attachmentOptionsDisplayed = newValue
                        )
```

Could it be like this? So we don't duplicate code for the current types we are handling currently?

Reply:
unfortunetly not, but thanks for the suggestion